base/system/winlogon/rpcserver.c

```diff
@@ -431,7 +448,12 @@ BaseAbortSystemShutdown(
     PREGISTRY_SERVER_NAME ServerName)
 {
     TRACE("BaseAbortSystemShutdown()\n");
-    return ERROR_SUCCESS;
+
+    //FIXME: Verify that the caller actually has the correct privileges
+
+    if (g_bShuttingDown == FALSE)
+        return ERROR_NO_SHUTDOWN_IN_PROGRESS;
+    return PostMessage(g_hShutdownDialog, WM_CLOSE, 0, 0) ? ERROR_SUCCESS : GetLastError();
```

Comment:
Yesterday I was thinking about whether it could be possible to have some kind of race condition where g_hShutdownDialog goes invalid during two concurrent calls of BaseAbortSystemShutdown().

Reply:
Maybe the abort function should set the bool to false instead of the dialog proc?

Reply:
Now that I think of it, you never reset g_hShutdownDialog to NULL. As for concurrent calls to BaseAbortSystemShutdown, even if g_hShutdownDialog  goes invalid, nothing serious happens. PostMessage would just fail.